src/v/cluster/types.h

```diff
@@ -1003,6 +1003,11 @@ struct remote_topic_properties
     operator<<(std::ostream&, const remote_topic_properties&);
 };
 
+struct remote_partition_properties {
```

Comment:
I don't like the name honestly. It suggests that these are per-partition properties, but they are the same for all partitions!